compress/src/main/c/Makefile

```diff
@@ -56,22 +56,27 @@ OBJS = \
 $(LIBNAME): $(OBJS)
 	$(CC) $(LD_FLAGS) -o $@ $(OBJS)
 
-build: $(LIBNAME) $(if $(OS:linux=),,$(if $(ARCH:x86_64=),,build32))
-
-build32:
-	$(MAKE) ARCH=i386 build
+buildAARCH64OnX86:
+	$(MAKE) CC=aarch64-linux-gnu-gcc ARCH=aarch64 build
+buildX86OnAARCH64:
+	$(MAKE) CC=x86_64-linux-gnu-gcc ARCH=x86_64 build
+build: $(LIBNAME) $(if $(OS:linux=),,$(if $(ARCH:$(REALARCH)=),,$(if $(ARCH:x86_64=),buildX86OnAARCH64,buildAARCH64OnX86)))
```

Comment:
Any chance this could be changed to use ifeq and ifneq instead of if with substitution like it done here?

It's super unclear that `if $(OS:linux=)` means `take OS, substitute "linux" with "", then evaluate if the string is non-empty`. I think this would be equivalent to:
```
build:
    $(LIBNAME)
    ifeq ($(OS),darwin)
        ifeq ($(ARCH),$(REALARCH))
            ifeq ($(ARCH),x86_64)
                buildAARCH64OnX86
            else
                buildX86OnAARCH64
            endif
```

Reply:
I was following the convention that was laid out before, but I agree the ifeq is more readable even if it's more verbose. I'll update it.

Reply:
I went with using variables instead of calling the individual targets, my initial goal was to go with the previous style but I don't think that was particularly easy to read. I also don't think there is a lot of value building just for the current platform as we expect to support AARCH64 for the foreseeable future.